android/java/org/chromium/chrome/browser/util/PictureInPictureUtils.java

```diff
@@ -0,0 +1,60 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.util;
+
+import android.app.AppOpsManager;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.net.Uri;
+import android.os.Build;
+
+import androidx.annotation.NonNull;
+
+/**
+ * Auxiliary class containing static methods that helps to determine if picture in picture mode is
+ * supported by device and enabled via permission settings.
+ */
+public class PictureInPictureUtils {
+    /** Returns {@code true} if picture in picture permission is enabled in app settings. */
+    public static boolean hasPictureInPicturePermissionEnabled(@NonNull final Context context) {
+        final AppOpsManager appOps =
+                (AppOpsManager) context.getSystemService(Context.APP_OPS_SERVICE);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            return appOps != null
+                    && appOps.unsafeCheckOpNoThrow(
+                                    AppOpsManager.OPSTR_PICTURE_IN_PICTURE,
+                                    android.os.Process.myUid(),
+                                    context.getPackageName())
+                            == AppOpsManager.MODE_ALLOWED;
+        } else {
+            //noinspection deprecation
+            return appOps != null
+                    && appOps.checkOpNoThrow(
+                                    AppOpsManager.OPSTR_PICTURE_IN_PICTURE,
+                                    android.os.Process.myUid(),
+                                    context.getPackageName())
+                            == AppOpsManager.MODE_ALLOWED;
+        }
+    }
+
+    /**
+     * Launches picture in picture permission settings screen. Note: this method should be called
+     * only when a device correctly supports picture in picture mode.
+     */
+    public static void launchPictureInPictureSettings(@NonNull final Context context) {
```

Comment:
this doesn't seem like it should be needed, upstream should already be handling this